main.go

```diff
@@ -348,6 +329,12 @@ func enterProvider(wg *sync.WaitGroup, p provider.Provider, mergerCH <-chan []st
 				log.Errorf("Provider(%s): Failed to execute with %d targets (%v): %v", p, len(output), output, err)
 			}
 
+		case <-time.After(resyncInterval):
+			log.Debugf("Provider(%s): resync to current state from cache: %v", resultCache)
+			if len(resultCache) == 0 {
+				continue
+			}
```

Comment:
Isn't this just a no-op?

Reply:
sigh, after refactoring forgot to add the updateOrCreate call

Reply:
fixed